[GPU Process] [FormControls] Add a ControlPart for ApplePayButton

shallawa opened the pull request:
#### a586299ede2ec5ded2e1d4495c5b743e2a696540
<pre>
[GPU Process] [FormControls] Add a ControlPart for ApplePayButton
<a href="https://bugs.webkit.org/show_bug.cgi?id=251416">https://bugs.webkit.org/show_bug.cgi?id=251416</a>
rdar://104850550

Reviewed by Aditya Keerthi.

Treat ApplePayButton as a ControlPart and draw it through GraphicsContext::drawControlPart().
Remove the special case for drawing the ApplePayButton as a SystemImage.

* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/editing/FontAttributeChanges.h:
* Source/WebCore/platform/graphics/cocoa/controls/ApplePayButtonCocoa.h: Added.
* Source/WebCore/platform/graphics/cocoa/controls/ApplePayButtonCocoa.mm: Copied from Source/WebCore/platform/graphics/mac/controls/ImageControlsButtonMac.mm.
(WebCore::ApplePayButtonCocoa::ApplePayButtonCocoa):
(WebCore::ApplePayButtonCocoa::draw):
* Source/WebCore/platform/graphics/cocoa/controls/ControlFactoryCocoa.h: Added.
* Source/WebCore/platform/graphics/cocoa/controls/ControlFactoryCocoa.mm: Copied from Source/WebCore/platform/graphics/mac/controls/ImageControlsButtonMac.mm.
(WebCore::ControlFactoryCocoa::createPlatformApplePayButton):
* Source/WebCore/platform/graphics/controls/ApplePayButtonPart.cpp: Added.
(WebCore::ApplePayButtonPart::create):
(WebCore::ApplePayButtonPart::ApplePayButtonPart):
(WebCore::ApplePayButtonPart::createPlatformControl):
* Source/WebCore/platform/graphics/controls/ApplePayButtonPart.h: Added.
(WebCore::ApplePayButtonPart::buttonType const):
(WebCore::ApplePayButtonPart::buttonStyle const):
(WebCore::ApplePayButtonPart::locale const):
* Source/WebCore/platform/graphics/controls/ControlFactory.h:
* Source/WebCore/platform/graphics/mac/AppKitControlSystemImage.h:
* Source/WebCore/platform/graphics/mac/controls/ControlFactoryMac.h:
* Source/WebCore/platform/graphics/mac/controls/ImageControlsButtonMac.mm:
* Source/WebCore/platform/graphics/mac/controls/SearchFieldCancelButtonMac.mm:
* Source/WebCore/platform/graphics/mac/controls/SearchFieldMac.mm:
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::createApplePayButtonPartForRenderer):
(WebCore::RenderTheme::createControlPart const):
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::canCreateControlPartForRenderer const):
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;ControlPart&gt;::encode):
(IPC::ArgumentCoder&lt;ControlPart&gt;::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/259652@main">https://commits.webkit.org/259652@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b67292729fe49409976199aa6d02952536bf0cb6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105507 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14607 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38414 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114766 "Built successfully") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/174912 "Failed to checkout and rebase branch from PR 9364") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/109407 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16016 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/5833 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97810 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/114610 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111264 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/12195 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95175 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/39679 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/94059 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26811 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/81374 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7888 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28167 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8238 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/4751 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14015 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47710 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6665 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9913 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (commented):
Is there a reason we're not removing `ApplePayButtonSystemImage.h`/`ApplePayButtonSystemImage.mm`?

shallawa committed:
[GPU Process] [FormControls] Add a ControlPart for ApplePayButton
https://bugs.webkit.org/show_bug.cgi?id=251416
rdar://104850550

Reviewed by Aditya Keerthi.

Treat ApplePayButton as a ControlPart and draw it through GraphicsContext::drawControlPart().
Remove the special case for drawing the ApplePayButton as a SystemImage.

* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/editing/FontAttributeChanges.h:
* Source/WebCore/platform/graphics/cocoa/controls/ApplePayButtonCocoa.h: Added.
* Source/WebCore/platform/graphics/cocoa/controls/ApplePayButtonCocoa.mm: Copied from Source/WebCore/platform/graphics/mac/controls/ImageControlsButtonMac.mm.
(WebCore::ApplePayButtonCocoa::ApplePayButtonCocoa):
(WebCore::ApplePayButtonCocoa::draw):
* Source/WebCore/platform/graphics/cocoa/controls/ControlFactoryCocoa.h: Added.
* Source/WebCore/platform/graphics/cocoa/controls/ControlFactoryCocoa.mm: Copied from Source/WebCore/platform/graphics/mac/controls/ImageControlsButtonMac.mm.
(WebCore::ControlFactoryCocoa::createPlatformApplePayButton):
* Source/WebCore/platform/graphics/controls/ApplePayButtonPart.cpp: Added.
(WebCore::ApplePayButtonPart::create):
(WebCore::ApplePayButtonPart::ApplePayButtonPart):
(WebCore::ApplePayButtonPart::createPlatformControl):
* Source/WebCore/platform/graphics/controls/ApplePayButtonPart.h: Added.
(WebCore::ApplePayButtonPart::buttonType const):
(WebCore::ApplePayButtonPart::buttonStyle const):
(WebCore::ApplePayButtonPart::locale const):
* Source/WebCore/platform/graphics/controls/ControlFactory.h:
* Source/WebCore/platform/graphics/mac/AppKitControlSystemImage.h:
* Source/WebCore/platform/graphics/mac/controls/ControlFactoryMac.h:
* Source/WebCore/platform/graphics/mac/controls/ImageControlsButtonMac.mm:
* Source/WebCore/platform/graphics/mac/controls/SearchFieldCancelButtonMac.mm:
* Source/WebCore/platform/graphics/mac/controls/SearchFieldMac.mm:
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::createApplePayButtonPartForRenderer):
(WebCore::RenderTheme::createControlPart const):
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::canCreateControlPartForRenderer const):
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder<ControlPart>::encode):
(IPC::ArgumentCoder<ControlPart>::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/259652@main
